lib/pages/timetable/view/events/event_view.dart

```diff
@@ -19,9 +19,12 @@ import 'package:flutter_feather_icons/flutter_feather_icons.dart';
 import 'package:provider/provider.dart';
 
 class EventView extends StatefulWidget {
-  const EventView({Key key, this.eventInstance}) : super(key: key);
-
+  const EventView({Key key, this.eventInstance, this.uniEvent})
+      : assert((eventInstance != null && uniEvent == null) ||
+            (eventInstance == null && uniEvent != null)),
```

Comment:
Nit: Please throw an actual ArgumentError explaining that only one of these should be specified, otherwise if someone messes it up accidentally it will be hard for them to tell what's wrong.